proto/config/config.proto

```diff
@@ -21,10 +21,6 @@ message Global {
 
   // If present, configures identity.
   IdentityContext identity_context = 4;
-
-  // If present, indicates that the Mutating Webhook Admission Controller should
-  // be configured to automatically inject proxies.
-  AutoInjectContext auto_inject_context = 6;
```

Comment:
i think we should keep this as a deprecated placeholder, so as not to break newer components/cli reading an older config?
```proto
AutoInjectContext auto_inject_context = 6 [deprecated=true];
```

Reply:
I thought it would be cleaner to just remove it, given the unmarshaller is set to ignore unknown fields, so it actually doesn't brake. You agree?

Reply:
My first thought is also to mark this as deprecated. Per the [grpc doc](https://developers.google.com/protocol-buffers/docs/proto3#reserved), if later on we assign field `6` to another field, it will break older clients which still depend on field `6` to be the `auto_inject_context`. If we really want to remove it, we can use `reserved` to ensure `6` doesn't get recycled.

Reply:
Field numbers are used to identify the fields in the protobuf message binary format, but we're not storing the config in the protobuf binary format; we're storing them in JSON, and we're only using protobuf to map the JSON into the go structs. From my understanding, such mapping is done by matching fields labels and order doesn't matter. One can test this by manually shifting order on some of the proxy config fields in the ConfigMap, restarting the injector, and checking the behavior is the same.

Reply:
After further discussion over Slack, we realized `linkerd inject` does receive that config via a binary message, so we gotta deprecate that field instead of just removing it.